base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -453,6 +463,40 @@ where B: BlockchainBackend + 'static
             return Ok(());
         }
 
+        if self.blockchain_db.bad_block_exists(block_hash).await? {
+            debug!(
+                target: LOG_TARGET,
+                "Block with hash `{}` already validated as a bad block",
+                block_hash.to_hex()
+            );
+            return Err(CommsInterfaceError::ChainStorageError(
+                ChainStorageError::ValidationError {
+                    source: ValidationError::BadBlockFound {
+                        hash: block_hash.to_hex(),
+                    },
+                },
+            ));
+        }
+
+        // lets check that the difficulty at least matches the min required difficulty
+        let constants = self.consensus_manager.consensus_constants(new_block.header.height);
+        let min_difficulty = constants.min_pow_difficulty(new_block.header.pow.pow_algo);
+        let achieved = match new_block.header.pow_algo() {
+            PowAlgorithm::RandomX => randomx_difficulty(&new_block.header, &self.randomx_factory)?,
+            PowAlgorithm::Sha3x => sha3x_difficulty(&new_block.header)?,
+        };
+        if achieved < min_difficulty {
+            self.blockchain_db
+                .add_bad_block(
+                    new_block.header.hash(),
+                    self.blockchain_db.get_chain_metadata().await?.height_of_longest_chain(),
+                )
+                .await?;
+            return Err(CommsInterfaceError::InvalidBlockHeader(
+                BlockHeaderValidationError::ProofOfWorkError(PowError::AchievedDifficultyBelowMin),
+            ));
+        }
+
```

Comment:
I think this should be a proper difficulty check with `DifficultyCalculator::check_achieved_and_target_difficulty`, as it is too easy to mine a block with minimum difficulty.

If this check has passed for a specific block we do not need to check it again when we do full validation.

Reply:
The problem with this is, we do not know what the actual target is at this point.
This is here to force the user to spend some time mining it. By making this take longer to create than validate, you make any attack here very expensive.